test/e2e/registration_service_test.go

```diff
@@ -39,15 +44,375 @@ func (s *registrationServiceTestSuite) TestLandingPageReachable() {
+		tokenErr := mp["error"]
+		require.IsType(s.T(), tokenErr, "")
+		require.Equal(s.T(), "found unknown authorization header:1223123123", tokenErr.(string))
+	})
+	s.Run("verify_get_signup_error_invalid_token", func() {
+		// Call signup endpoint with an invalid token.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/signup", nil)
+		require.NoError(s.T(), err)
+		req.Header.Set("Authorization", "Bearer 1223123123")
+		req.Header.Set("content-type", "application/json")
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+
+		// Retrieve unauthorized http status code.
+		assert.Equal(s.T(), http.StatusUnauthorized, resp.StatusCode)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		// Check token error.
+		tokenErr := mp["error"]
+		require.IsType(s.T(), tokenErr, "")
+		require.Equal(s.T(), "token contains an invalid number of segments", tokenErr.(string))
+	})
+	s.Run("verify_get_signup_valid_token_status_200OK", func() {
+		// Call signup endpoint with an valid token.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/signup", nil)
+		require.NoError(s.T(), err)
+		req.Header.Set("Authorization", "Bearer "+token0)
+		req.Header.Set("content-type", "application/json")
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		assert.Equal(s.T(), http.StatusOK, resp.StatusCode)
+	})
+	s.Run("verify_get_signup_valid_token_status_404NotFound", func() {
+		// Get valid generated token for e2e tests. IAT claim is overriden
+		// to avoid token used before issued error.
+		identity1 := authsupport.NewIdentity()
+		emailClaim1 := authsupport.WithEmailClaim(uuid.NewV4().String() + "@email.tld")
+		iatClaim1 := authsupport.WithIATClaim(time.Now().Add(-60 * time.Second))
+
+		// Not identical to the token used in POST signup - should return resource not found.
+		token1, err := authsupport.GenerateSignedE2ETestToken(*identity1, emailClaim1, iatClaim1)
+		require.NoError(s.T(), err)
+
+		// Call signup endpoint with an valid token.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/signup", nil)
+		require.NoError(s.T(), err)
+		req.Header.Set("Authorization", "Bearer "+token1)
+		req.Header.Set("content-type", "application/json")
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		assert.Equal(s.T(), http.StatusNotFound, resp.StatusCode)
+	})
+}
```

Comment:
could you add a test where the token is expired? (that may require a `WithExpiryClaim` option to pass when initializing the token)

Reply:
added here: b2f827249f349d2886f4f1fa5400522803dfbce1

Reply:
thanks!